src/metabase/api/common.clj

```diff
@@ -491,11 +491,16 @@
 ;;; ------------------------------------------ PARAM PARSING FNS ----------------------------------------
 
 (defn bit->boolean
-  "Coerce a bit returned by some MySQL/MariaDB versions in some situations to Boolean."
+  "Coerce a bit returned by some MySQL/MariaDB versions in some situations to Boolean.
+
+  MariaDB is especially strange: https://stackoverflow.com/questions/78466426/bit1-in-view-using-conditionals-in-mariadb-returns-48-49"
   [v]
-  (if (number? v)
-    (not (zero? v))
-    v))
+  (cond
+    (number? v)                      (not (zero? v))
+    (and (bytes? v) (= (count v) 1)) (case (char (first v))
```

Comment:
Pretty sure you can do `zero?` on a byte

```clj
> (zero? (byte 0))
true
> (zero? (byte 1))
false
```

Reply:
Instead of `(case (char (first v)) ...)` I'm pretty sure you can just do `(not (zero? (first v)))`

Reply:
We aren't looking for zero, we are looking for 48 or 49. I added the conversion to char to make it more obvious the hell is going on.